### ISSUE TITLE
How to execute with arguments?

### ISSUE THREAD

**Issue by ehannes** [crismali/fantaskspec#10]
Is it possible to execute a task with arguments? I tried `task.execute(args)` without success.

**Comment by crismali** [comment]
`task.execute("foo", "bar", "baz")` works when I test it. How is your task set up? Here's mine for reference:

```ruby
desc "Some rake task"
task :some_task do |t, args|
  "I don't do much"
  puts "args: #{args}"
end

# later...
task.execute("foo", "bar", "baz")
```

**Comment by ehannes** [comment]
I had my task set up like this:
```
desc "Some rake task"
task :some_task, [:arg1, :arg2] do |task, args|
  p args[:arg1]
  p args[:arg2]
end
```

I'll try your approach and see if it helps :)

**Comment by ehannes** [comment]
WIth your approach, `args` is a `String` object. With my approach, `args` is a `Rake::TaskArguments` object. With the `TaskArguments` object, you can do things like `args[:arg1]` as in my example above. And when you run the task from the terminal, you also get a `TaskArguments` object if you print the object:

```
$ rake some_task['foo', 'bar']
args: #<Rake::TaskArguments args1: foo, args2: bar>
```

But I'm new to Rake tasks. Maybe there is a way to handle this better? Or test this with this gem.

**Comment by crismali** [comment]
@ehannes At the moment there isn't anything baked into the gem to help with this, but until that happens you can create the `Rake::TaskArguments` object yourself and then pass it to `execute`

```ruby
args = Rake::TaskArguments.new([:arg1, :arg2], ["foo", "bar"])
task.execute(args)
```

**Comment by ehannes** [comment]
@crismali Yes, that was what I ended up doing. Would it be possible to somehow bake this into the gem?

**Comment by crismali** [comment]
@ehannes Sure. I'd probably define some helper like `to_task_arguments` that would be able to figure that out. Hopefully I'll have some time to try that soon, but for now PRs are always welcome 👍

**Comment by ehannes** [comment]
I have no clue of how to implement such a thing. Otherwise, I would gladly open a PR.

**Comment by crismali** [comment]
@ehannes just published version 1.1 which has the `to_task_arguments` helper. Hope it helps! 👍
